components/admin/lessons/AdminLessonSideNav.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect } from 'react'
+import { ApolloError } from '@apollo/client'
+import { AlertFillIcon, ChevronRightIcon } from '@primer/octicons-react'
+import { isNumber } from 'lodash'
+import { Spinner } from 'react-bootstrap'
+import { get } from 'lodash'
+import styles from '../../../scss/adminLessonSideNav.module.scss'
+
+type Item = {
+  id: number
+  name: string
+  content: string
+  lesson: { id?: number }
+}
+
+type Items = Item[]
+
+type Props = {
+  items: Items | undefined
+  title: string
+  lessonId: number
+  onAddItem: (m: Item | undefined) => void
+  loading?: boolean
+  error?: ApolloError
+  setActive?: React.Dispatch<React.SetStateAction<number>>
+  active?: number
+}
+
+const AdminLessonSideNav = ({
+  items,
+  onAddItem,
+  loading,
+  error,
+  setActive,
+  active,
+  title,
+  lessonId
+}: Props) => {
+  const setAsActive = (id: number) => setActive && setActive(id)
+
+  const itemsList =
+    items &&
+    items.reduce((acc: JSX.Element[], item) => {
+      if (item.lesson.id === lessonId) {
+        const isActive = active === item.id
+        const className =
+          styles[
+            isActive
+              ? 'container__modulesList__module--active'
+              : 'container__modulesList__module'
+          ]
+        const withChevron = !isActive && <ChevronRightIcon size={16} />
+
+        acc.push(
+          <li
+            className={className}
+            key={item.id}
+            onClick={() => setAsActive(item.id)}
+          >
+            <span>{item.name}</span>
+            {withChevron}
+          </li>
+        )
+      }
+
+      return acc
+    }, [])
+
+  useEffect(() => setActive && setActive(-1), [lessonId])
```

Comment:
why do we need `useEffect` here?

Reply:
When the current lesson changes, I think it's better not to set any active item by setting its value as `-1`. By setting the value to `-1`, it'll be in add-new-module mode.

Reply:
I think its better for the parent to be managing the state, the parent should be passing in `-1`. Then we don't have to do this right? I'm imaginging here's how the parent would use this:

```js

const AdminModulePage = () => {
  const [selected, setSelected] => useState(-1)
  
  const items = [{
    title: 'one',
  }, {title: 'two'}]
  
  return (
    <AdminLessonSideNav
      onClick = {(index) => setSelected(index)}
      selectedIndex= {selected}
    />
  )
 ```